docs/guides/developer/docs/reviewing-and-merging.md

```diff
@@ -44,21 +43,19 @@ all pull requests are to ensure that:
     * [Licenses seem to be fine](license.md)
     * Licenses are added to the `NOTICES` file
 * Feature documentation is in place
-* An upgrade path exists (usually relevant for database changes)
-
-
-The exact review process heavily depends on the type, size and purpose of the pull request.
+* Adopters can easily upgrade to new versions
+    * An upgrade path exists
+    * The upgrade path is documented
 
 Here are some things a reviewer should usually do:
 
 * Check if code looks sensible (e.g. no nonsensical files, no obvious formatting errors, no large binary files, …)
 * Locally run Opencast and verify that the issue has been fixed (as described in the pull request and/or the Jira
   ticket) or the feature does what it is supposed
+* Check whether the database schema is modified. In case it is, a database migration scripts needs to be provided
+  and the upgrade guide needs to contain the information that a database migration is needed including a reference
+  to the Jira ticket(s) the cause the migration
```

Comment:
-1 to just referencing Jira. It depends on the change but mostly, I would expect a description of the upgrade path, not a list of changes. That's part of the changelog. just removing the “including a reference to the Jira ticket(s) the cause the migration” should do.

Reply:
Let's discuss this on list ;-)

Reply:
Suggested change:

> If the database schema is modified, check if a database migration script is provided and the upgrade guide contains migration documentation.

Reply:
Very much agree with a descriptive fix, rather than linking to JIRA.  JIRA is where highly technical discussions can take place.  Adopters neither need to see it, nor in many cases even want to know about it.  They only want the quick-hit fix so they an get on with their lives.  I'm fine including a link as well, but it needs to be more in the vein of "for more information see MH-Whatever".

Reply:
@gregorydlogan I very much thought of the "for more information see MH-XXXXX". That does make sense as it might sometimes not be clear how the value needs to be adjusted an the issue might be helpful to get more information about the change.
Note that the changelog is not providing this information as the issue titles might not always obviously relate to the configuration change.

Reply:
The values should always have proper documentation right at the key in the documentation file. For me, that's by far the most important part of any configuration change.

But note that I'm not saying that you may not link Jira if it makes sense or a pull request, a discussion on list or any other documentation. If that is helpful, please do! If it's not …don't.

Reply:
I'm not talking about the documentation of the configuration which I agree should be in the configuration file right were the configuration settings is located.

The reference to the ticket is supposed to provide more context about the configuration change in means of "What was changed in Opencast that caused the configuration to be changed" (not: how do I have to set a value) in case somebody want to dig deeper / have more information.

Always referring to the ticket simply avoids discussions whether a reference to the ticket makes sense for a given case. Any information additionally useful can be referenced from the ticket (rather than the upgrade guide). This includes pull requests, discussions on list and whatever might be relevant.

Reply:
Referring to the ticket is great, requiring that an adopter read a ticket is not.  If the config and the markdown docs don't provide enough explanation of how a value *should* be set, then there needs to be additional work done on those files.  Putting the 'what' into the config and docs is required, the 'why' can be in both, but I'm betting most adopters are more interested in 'what' they should be setting rather than 'why'.

tl;dr: An adopter should never be dropped into JIRA.  It's enough of a mess as it is, and we've got docs which sometimes are not in sync with what's discussed in JIRA.  We don't need another source of truth for adopters.

Reply:
@gregorydlogan 

This example quite nails what I mean:

> - Added org.opencastproject.elasticsearch.server.address & org.opencastproject.elasticsearch.server.port because of added support for external elastic search servers([MH-13253](https://opencast.jira.com/browse/MH-13253))